WooCommerce/UITestsFoundation/Screens/Products/SingleProductScreen.swift

```diff
@@ -41,17 +41,32 @@ public final class SingleProductScreen: ScreenObject {
         return self
     }
 
-    public func verifyNewProductScreenLoaded(productName: String) {
+    public func verifyPublishedProductScreenLoaded(productType: String, productName: String) {
+        let priceLabel = NSPredicate(format: "label == 'Price'")
+        let addVariationLabel = NSPredicate(format: "label == 'Add variations'")
+
+        // common fields on a published product screen
         XCTAssertTrue(app.buttons["save-product-button"].exists)
         XCTAssertTrue(app.staticTexts["TIP"].exists)
         XCTAssertTrue(app.textViews[productName].exists)
+
+        // different product types displays different fields on the published product screen
+        // this is to validate that the correct screens are displayed
+        switch productType {
+        case "physical", "virtual":
+            XCTAssertTrue(app.staticTexts.containing(priceLabel).firstMatch.exists)
+        case "variable":
+            XCTAssertTrue(app.staticTexts.containing(addVariationLabel).firstMatch.exists)
+        default:
+            fatalError("Product Type \(productType) doesn't exist!")
```

Comment:
Have you considered failing the test with `XCTFail` instead of crashing here?

Also, and this might be a change for a future PR, have you considered using an `enum` to encode the possible product types instead of a `String`? Doing so would make it impossible to run into this scenario. 

I'm a fan of the concept of **making undesirable state unrepresentable**. Evan Czaplicki, the author of the delightful programming language Elm, puts is better than I can in [this Appendix to the Elm documentation](https://https://guide.elm-lang.org/appendix/types_as_sets.html#who-cares), which I really recommend reading in full:

> So the whole point here is that ruling out invalid data makes your code shorter, simpler, and more reliable. By making sure the set of possible values in code exactly matches the set of valid values in real life, many problems just go away

Reply:
It looks like 1b22d2f2bcb9b9bf340c7172bef3eb6927e70eb0 applies the `fatalError` ~> `XCTFail` change only for the L85, while on L65 the `fatalError` is still used.

Reply:
missed this since it was existing code, but makes sense to update this in this PR. updated in https://github.com/woocommerce/woocommerce-ios/pull/9441/commits/44d8ab4ce93395b6af9387bc677a31c3ae7ee9c6